cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/ssa0/SsaInternals.qll

```diff
@@ -208,11 +208,24 @@ class Phi extends TPhi, SsaDefOrUse {
   final override PhiNode asPhi() { result = phi }
 
   final override Location getLocation() { result = phi.getBasicBlock().getLocation() }
+
+  Definition getAnInput() { Cached::lastRefRedef(result, _, _, phi) }
+}
+
+private Node0Impl getAnUltimateValueForDef(Definition def) {
+  exists(IRBlock bb, int i, SourceVariable sv | def.definesAt(sv, bb, i) |
+    result = getAnUltimateValueForDef(any(Phi phi | phi.asPhi() = def).getAnInput())
```

Comment:
Why does the exists wrap this case? The only reason I can think of is that we want to be sure we're actually in a basic block.

Reply:
Well spotted! I've fixed this in feea2a71c2b36b57e1b958f4a4ea7e6447782333 (as well as simplified the predicate a bit more). Since this is part of SSA we _should_ always have a basic block (since the SSA internals depend on always having a basic block).